app/src/main/java/org/zalando/nakadi/config/NakadiConfig.java

```diff
@@ -47,14 +48,26 @@ public void run() {
         return taskExecutor;
     }
 
+    @Bean
+    @Qualifier("pre-internal-publishing-checks")
```

Comment:
```suggestion
    @Qualifier("internal-publishing-checks")
```

Reads less awkward this way, IMO.  Maybe change the other to just "publishing-checks" as well, there are only checks prior to publishing anyway.